org.opentosca.planbuilder.core.bpel/src/org/opentosca/planbuilder/core/bpel/typebasedplanbuilder/BPELSituationAwareBuildProcessBuilder.java

```diff
@@ -0,0 +1,320 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.SAXException;
+
+/**
+ * <p>
+ * This Class represents the high-level algorithm of the concept in <a href=
+ * "http://www2.informatik.uni-stuttgart.de/cgi-bin/NCSTRL/NCSTRL_view.pl?id=BCLR-0043&mod=0&engl=1&inst=FAK"
+ * >Konzept und Implementierung eine Java-Komponente zur Generierung von WS-BPEL 2.0 BuildPlans fuer
+ * OpenTOSCA</a>. It is responsible for generating the Build Plan Skeleton and assign plugins to
+ * handle the different templates inside a TopologyTemplate.
+ * </p>
+ *
+ * Copyright 2013 IAAS University of Stuttgart <br>
+ * <br>
+ *
+ * @author Kalman Kepes - kepeskn@studi.informatik.uni-stuttgart.de
+ *
+ */
+public class BPELSituationAwareBuildProcessBuilder extends AbstractBuildPlanBuilder {
+
+    final static Logger LOG = LoggerFactory.getLogger(BPELSituationAwareBuildProcessBuilder.class);
+
+    // class for initializing properties inside the plan
+    private final PropertyVariableHandler propertyInitializer;
+    // class for initializing output with boundarydefinitions of a
+    // serviceTemplate
+    private final ServiceTemplateBoundaryPropertyMappingsToOutputHandler propertyOutputInitializer;
+    // adds serviceInstance Variable and instanceDataAPIUrl to buildPlans
+
+    private SimplePlanBuilderServiceInstanceHandler serviceInstanceInitializer;
+
+    private CorrelationIDInitializer correlationHandler;
+
+    private SituationTriggerRegistration sitRegistrationPlugin;
+
+    // class for finalizing build plans (e.g when some template didn't receive
+    // some provisioning logic and they must be filled with empty elements)
+    private final BPELFinalizer finalizer;
+
+    private BPELPlanHandler planHandler;
+
+    private BPELPluginHandler bpelPluginHandler = new BPELPluginHandler();
+
+    private NodeRelationInstanceVariablesHandler nodeRelationInstanceHandler;
+
+    private final EmptyPropertyToInputHandler emptyPropInit = new EmptyPropertyToInputHandler();
+
+    private BPELProcessFragments fragments;
+
+    /**
+     * <p>
+     * Default Constructor
+     * </p>
+     */
+    public BPELSituationAwareBuildProcessBuilder() {
+        try {
+            this.planHandler = new BPELPlanHandler();
+            this.serviceInstanceInitializer = new SimplePlanBuilderServiceInstanceHandler();
+            this.nodeRelationInstanceHandler = new NodeRelationInstanceVariablesHandler(this.planHandler);
+            this.sitRegistrationPlugin = new SituationTriggerRegistration();
+            this.correlationHandler = new CorrelationIDInitializer();
+            this.fragments = new BPELProcessFragments();
+
+        }
+        catch (final ParserConfigurationException e) {
+            BPELSituationAwareBuildProcessBuilder.LOG.error("Error while initializing BuildPlanHandler", e);
+        }
+        // TODO seems ugly
+        this.propertyInitializer = new PropertyVariableHandler(this.planHandler);
```

Comment:
you're not checking that the `planHandler` has in fact been initialized... could that be dangerous?

Reply:
Should now initialize only when the planhandler is properly initialized